XBOXOLD.cpp

```diff
@@ -293,6 +293,19 @@ void XBOXOLD::printReport(uint16_t length __attribute__((unused))) { //Uncomment
 }
 
 int8_t XBOXOLD::getAnalogIndex(ButtonEnum b) {
+        // For legacy reasons these mapping indices not match up,
+        // as the original code uses L1/R1 for the triggers and
+        // L2/R2 for the white/black buttons. To fix these new enums
+        // we have to transpose the keys before passing them through
+        // the button index function
+        switch (b) {
+        case(LT): b = L1; break;  // normally L2
+        case(RT): b = R1; break;  // normally R2
```

Comment:
Shouldn't this be `L2` and `R2` or have I misunderstood it?

Reply:
It's correct, it just looks a little odd. My comment probably could have explained it better.

The library's implementation of the original Xbox controller is a little wonky because of the shared enum IDs. The controller only has one set of triggers so those were programmed as `L1/R1`, and the white/black buttons that became `LB/RB` on the 360 controller were programmed with the same IDs as `L2/R2`. That creates a conflict because the other Xbox controllers use the reverse - `L2/R2` for the triggers and `L1/R1` for the option buttons (bumpers).

Without this little filter that means `LT/RT` returns the white/black buttons, and `LB/RB` return the triggers. The filter swaps the enum values around before it runs through the ID filter, so whether `L1` is passed (previous versions) or `LT` is passed (current versions) the trigger is still outputted.